exonum/src/storage/keys.rs

```diff
@@ -388,4 +410,86 @@ mod tests {
         // Notice the different order of values compared to the previous test
         assert_eq!(index.values().collect::<Vec<_>>(), vec![100, 200]);
     }
+
+    #[test]
+    fn test_storage_key_for_system_time_round_trip() {
+        use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+        let times = [
```

Comment:
I think we should test for the other extreme compared to `UNIX_EPOCH`, and extreme values of nanoseconds.